libs/types/src/domain_address.rs

```diff
@@ -10,113 +10,120 @@
 // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 // GNU General Public License for more details.
 
-use cfg_utils::vec_to_fixed_array;
 use frame_support::pallet_prelude::RuntimeDebug;
 use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
-use sp_runtime::traits::AccountIdConversion;
+use sp_runtime::{traits::AccountIdConversion, TypeId};
 
 use crate::EVMChainId;
 
+const MAX_ADDRESS_SIZE: usize = 32;
+pub type LocalAddress = [u8; 32];
+pub type EthAddress = [u8; 20];
+
+pub fn local_to_eth_address(address: LocalAddress) -> EthAddress {
```

Comment:
Nit: Let's stick to one namespace for Eth, Evm, Solidity
```suggestion
pub fn local_to_evm_address(address: LocalAddress) -> EthAddress {
```

Reply:
I wanted to distinguish for the name: 
- eth = just the address
- evm: and ethereum address + chain information

I say this because not sure for example if `[u8; 20]` is an evm address, I think not because it does not have chain information 🤔 

But not sure if this association makes sense TBH. It seems not 😅

Reply:
I definitely get what you are saying but to me, having two namespaces just causes confusion.

However, since he address format is officially referred to as the _Ethereum address format_, it appears to me your distinction is correct. So fine with keeping as is.

Reply:
Will add a comment regarding this then to explain the why. Thanks for the ref

Reply:
Frontier's approach is pretty straightforward when it comes to eth/evm addresses, they name these `AccountId20` - [here](https://github.com/polkadot-evm/frontier/blob/master/primitives/account/src/lib.rs#L42). 

As someone who didn't have to deal much with `Domain`/`DomainAddress`, I find that local/eth/evm are confusing as well tbh.

Why not go for something simpler like:
- `CentrifugeAccount` - `AccountId32`.
- `EvmAccount` - `AccountId20`/`[u8;20]`.
- `CentrifugeEvmAccount` - `AccountId32` that's basically `EvmAccount` + `chain_id` + `tag`.

Then we could have implementations for:
- `CentrifugeAccount` -> `EvmAccount` - truncate;
- `EvmAccount` -> `CentrifugeAccount` - pad with 0s;
- `CentrifugeAccount` <-> `CentrifugeEvmAccount` - no need;
- `EvmAccount` -> `CentrifugeEvmAccount` - append `chain_id` + `tag`;
- `CentrifugeEvmAccount` -> `EvmAccount` - truncate;

Reply:
Thanks for sharing this!

And these `CentrifugeAccount,` etc, should be types on their own, right? So you would directly remove the `DomainAddress` type? Which type will you use to track all of them generically?

Reply:
No, I would add them to the `DomainAddress`:

```
pub enum DomainAddress {
	/// A standard Centrifuge account.
	Centrifuge(CentrifugeAccount),
	
	/// A Centrifuge account derived from an `EvmAccount`.
	CentrifugeEvm(CentrifugeEvmAccount),
	
	/// An EVM-compatible account.
	Evm(EvmAccount),
}
```

This way we can keep the above mentioned conversions, and just have `From` implementations for `DomainAddress`, and for each of these account types.

Reply:
Ok, I like the way you're taking it. 

Just a problem with this. The variant `Evm(EvmAccount)` will not have a `Domain` information associated with it. A DomainAddress requires it to later extract this.

For sure, I would not have time to do such a refactor for Monday 😅

Reply:
We could add a struct for the evm account that has the chain and the actual account, or keep it as it is right now? I dunno, just thinking out loud at this point ^^

Reply:
But that would be what we have, right? 🤔 

I'm going to sketch a new draft with some ideas from this, but reusing the current `DomainAddress`, let's see. Thanks for your thoughts!